src/index.js

```diff
@@ -51,24 +68,31 @@ function syncReduxAndRouter(history, store, selectRouterState = SELECT_STATE) {
   }
 
   const unsubscribeHistory = history.listen(location => {
-    const routePath = locationToString(location);
+    const route = {
+      path: history.createPath(location),
+      state: location.state
+    };
+
+    // Avoid dispatching an action if the store is already up-to-date,
+    // even if `history` wouldn't do anything if the location is the same
+    if(locationsAreEqual(getRouterState(), route)) return;
```

Comment:
I'm not sure we actually even need this check anymore. The only reason is existed was to stop the cyclical update between redux and the router. But now that cycle is broken with the `changeId`. The `avoidRouterUpdate` flag below makes the reducer not increment it, so the redux listener will not call back in here.

I think we can just remove it, and always fire a `pushPath` call now, yes?